### DISCUSSION TITLE
Returning self instead of None for Widget "verb" methods

### DISCUSSION THREAD

**Discussion by liancheng** [Textualize/textual#1817] [Ideas]
My app has a form-like interface, and binds one key to each widget to set focus and optionally adjust the input, e.g., toggling a switch, moving the cursor to the next `ListItem` in a `ListView`, etc.

For `Switch`es, I wrote the following method to achieve this:

```python
def action_toggle_switch(self, selector: str) -> None:
    switch = self.query_one(selector, Switch)
    switch.focus()
    switch.toggle()
```

The fact that both `focus()` and `toggle()` return `None` prevented me from chaining the calls. If they returned `self` instead, the above method can be simplified to:

```python
def action_toggle_switch(self, selector: str) -> None:
    self.query_one(selector, Switch).focus().toggle()
```

I'm not quite sure whether this violates certain well-established Python best practices or conventions, though.

**Comment by davep** [comment]
Chatting about it here, the feeling seems to be that this would be a good idea (it's one I'm a big fan of coming from other environments where it's generally less frowned upon). I'll raise a ticket for this and add it to the backlog.

**Comment by rodrigogiraoserrao** [comment]
Hey @liancheng , you'll be happy to know that some widget verb methods return `self` as of #2102.
See the linked PR or the Changelog for a list of methods that support this!

**Reply by liancheng** [reply]
Woohoo! Thank you!
